### PR TITLE
Tag code samples for usage in developer docs

### PR DESCRIPTION
**Pull request by cbetta**
I've tagged the majority of code samples to an endpoint for usage in the new developer documentation.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @cbetta has signed the CLA. Thanks for the pull request!

**Comment by carycheng** [comment]
Will take a second look at this, before merging, thank you! @cbetta

**Comment by cbetta** [comment]
Thanks @carycheng. Just FYI, some of the samples might not be quite the right ones, but the current plan is to review the documentation resource by resource with the devrel team and engineering teams, and improving the samples along the way.

**Comment by cbetta** [comment]
@carycheng @mattwiller I am happy to merge this as-is for now.

File: doc/files.md [hunk 1]
```diff
@@ -646,6 +658,7 @@ Metadata can be created on a file by calling
 Note: This method will only succeed if the provided metadata template is not currently applied to the file, otherwise
 it will fail with a Conflict error.
 
+<!-- sample post_files_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/files.md]: maybe post_files_files_id_metadata_scope_template for the url?

File: doc/files.md [hunk 2]
```diff
@@ -654,6 +667,7 @@ file.createMetadata(new Metadata().add("/foo", "bar"));
 
 Update a files Metadata by calling [`updateMetadata(Metadata properties)`][update-metadata].
 
+<!-- sample put_files_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/files.md]: put_files_files_id_metadata_scope_template

File: doc/files.md [hunk 3]
```diff
@@ -677,6 +691,7 @@ Retrieve a files Metadata by calling [`getMetadata()`][get-metadata],
 [`getMetadata(String templateKey, String templateScope)`][get-metadata-3].
 These methods return a [`Metadata`][metadata] object, which allows access to metadata values.
 
+<!-- sample get_files_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/files.md]: get_files_files_id_metadata_scope_template

File: doc/files.md [hunk 4]
```diff
@@ -723,6 +738,7 @@ A files Metadata can be deleted by calling
 [`deleteMetadata(String templateKey)`][delete-metadata-2], or
 [`deleteMetadata(String templateKey, String templateScope)`][delete-metadata-3].
 
+<!-- sample delete_files_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/files.md]: delete_files_files_id_metadata_scope_template

File: doc/folders.md [hunk 5]
```diff
@@ -320,6 +327,7 @@ Metadata can be created on a folder by calling
 Note: This method will only succeed if the provided metadata template is not currently applied to the folder, otherwise
 it will fail with a Conflict error.
 
+<!-- sample post_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/folders.md]: Similar to the files/metadata endpoints I'm not sure if you wanted to go by url?

File: doc/folders.md [hunk 6]
```diff
@@ -343,7 +351,7 @@ Retrieve a folder's metadata by calling [`getMetadata()`][get-metadata],
 [`getMetadata(String templateKey, String templateScope)`][get-metadata-3].
 These methods return a [`Metadata`][metadata] object, which allows access to metadata values.
 
-
+<!-- sample get_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/folders.md]: get_folders_folders_id_metadata_scope_template

File: doc/folders.md [hunk 7]
```diff
@@ -371,6 +379,7 @@ Update Metadata
 
 Update a folder's metadata by calling [`updateMetadata(Metadata properties)`][update-metadata].
 
+<!-- sample put_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/folders.md]: put_folders_folders_id_metadata_scope_template

File: doc/folders.md [hunk 8]
```diff
@@ -386,6 +395,7 @@ A folder's metadata can be deleted by calling
 [`deleteMetadata(String templateKey)`][delete-metadata-2], or
 [`deleteMetadata(String templateKey, String templateScope)`][delete-metadata-3].
 
+<!-- sample delete_folders_id_metadata_id_id -->
```
Comment (carycheng) [inline doc/folders.md]: delete_folders_folders_id_metadata_scope_template

**Comment by mattwiller** [comment]
@carycheng The format we're using for these comment tags just replaces every path parameter with `id` for now, so the metadata tags you mentioned are correct as far as the docs build system is concerned.

**Pull request merged**